db/dcrpg/pgblockchain.go

```diff
@@ -1113,6 +1113,13 @@ func (pgb *ChainDB) VotesInBlock(hash string) (int16, error) {
 	return voters, nil
 }
 
+// piparser panics should not cause dcrdata panics.
+func recoverPiparser() {
+	if r := recover(); r != nil {
+		log.Errorf("recovered from piparser panic: %v", r)
+	}
+}
```

Comment:
I'm tempted to say relaunch `proposalsUpdateHandler`, but it's possible or even likely to panic again depending on the root cause.  Perhaps relaunch after a short delay.  Thoughts?

Reply:
That sounds good. Should we print the `debug.Stack()` ?

Reply:
Ah, yeah, totally.